### PR TITLE
Add stack module

### PR DESCRIPTION
**Pull request by briandowns**
Add Stack Module

Resolves: #

### What's Changed:

Adds a new module that implements a stack and includes tests and docs.

### Type of Change:

- [ ] Bug fix
- [ x ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).
- [ ] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: docs/docs/standard-lib/stack.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Adds the given value to the stack and resizes the stack if necessary.
+
+```cs
+const stack = Stack.new();
+stack.push("some_value");
+```
+
+### Stack.pop() -> Value
+
+Returns the value at the top of the stack. The capacity will be decreased if necessary.
```
Comment (Jason2605) [inline docs/docs/standard-lib/stack.md:58]:
```suggestion
Returns the value at the top of the stack. The capacity will be decreased if necessary.

Note: A runtime error will occur if there is an attempt to pop on an empty stack
```

**Review by Jason2605 — APPROVED** [review]
Small comment on the docs other than that looks neat 👍

**Comment by briandowns** [comment]
Al set!

**Comment by Jason2605** [comment]
Hah the new script is failing 😅

**Comment by Jason2605** [comment]
Thanks for this!

**Pull request merged**